logstash-rs/src/buffer.rs

```diff
@@ -39,18 +41,26 @@ impl BufferedSender {
 
 impl Sender for BufferedSender {
     fn send(&self, event: LogStashRecord) -> Result<()> {
-        self.sender.send(Command::Send(event)).map_err(From::from)
+        let result = self.sender.try_send(Command::Send(event));
+        process_result(result)
     }
 
     fn send_batch(&self, events: Vec<LogStashRecord>) -> Result<()> {
-        self.sender.send(Command::SendBatch(events))?;
-        Ok(())
+        let result = self.sender.try_send(Command::SendBatch(events));
+        process_result(result)
     }
 
     fn flush(&self) -> Result<()> {
-        self.sender.send(Command::Flush)?;
-        Ok(())
+        let result = self.sender.try_send(Command::Flush);
+        process_result(result)
+    }
+}
+
+fn process_result<T>(r: std::result::Result<(), TrySendError<T>>) -> Result<()> {
+    if matches!(r, Err(TrySendError::Disconnected(..))) {
```

Comment:
I think it would be good to return error also on overflow. But to reduce the error intensity (by default log4rs write them to stderr) do that only for messages on Warn level and above

Reply:
Done for either single message or batch with at least one `Warn`